remove out of date information on device memory management (#488)
ROCm/rocm-libraries: projects/rocblas/docs/reference/memory-alloc.rst

********************************************************************

For temporary device memory, rocBLAS uses a per-handle memory allocation with out-of-band management.

> More information is in the Device memory allocation section of :ref:programmers-guide.

Reviewer: ```suggestion
For more information, see the device memory allocation section of the :ref:programmers-guide.
```
